### PR TITLE
netaddr: add IP.IsPrivate

### PR DESCRIPTION
**Pull request by mdlayher**
Signed-off-by: Matt Layher <mdlayher@gmail.com>

This is probably the last method I need for a bit. Notably this mirrors the upcoming https://tip.golang.org/pkg/net/#IP.IsPrivate for Go 1.17.

### CODE REVIEW

File: netaddr.go [hunk 1]
```diff
@@ -666,6 +666,28 @@ func (ip IP) IsGlobalUnicast() bool {
...
+		// RFC 1918 allocates 10.0.0.0/8, 172.16.0.0/12, and 192.168.0.0/16 as
+		// private IPv4 address subnets.
+		return ip.v4(0) == 10 ||
+			(ip.v4(0) == 172 && ip.v4(1)&0xf0 == 16) ||
+			(ip.v4(0) == 192 && ip.v4(1) == 168)
+	}
+
+	// RFC 4193 allocates fc00::/7 as the unique local unicast IPv6 address
+	// subnet.
+	return ip.v6(0)&0xfe == 0xfc
```
Comment (bradfitz) [inline netaddr.go:688]: if ip.Is6() {
     return ...
}

return false // IP zero value

**Comment by mdlayher** [comment]
Updated, PTAL! Thanks.

File: netaddr.go [hunk 2]
```diff
@@ -666,6 +666,27 @@ func (ip IP) IsGlobalUnicast() bool {
 		!ip.IsLinkLocalUnicast()
 }
 
+// IsPrivate reports whether ip is a private address, according to RFC 1918
+// (IPv4 addresses) and RFC 4193 (IPv6 addresses).
```
Comment (bradfitz) [inline netaddr.go:670]: // ... That is, it reports whether ip is in 10.0.0.0/8, 192.168.0.0/16, .... or fc00::/7. This is the same
// as the standard library's net.IP.IsPrivate.

**Comment by mdlayher** [comment]
Updated docs.

**Review by bradfitz — APPROVED** [review]

**Pull request merged**